enhancements/olm/olm-managed-operator-metrics.md

```diff
@@ -0,0 +1,129 @@
+
+> Note: The `InstallPlan` will include a helpful message for cluster admins if either of the situations above are encountered.
+
+On upstream clusters, there is no way for OLM to identify which Prometheus Operator ServiceAccount should be granted the appropriate RBAC privileges to view the `ServiceMonitor` and `PrometheusRule` resource events in the namespace that the operator is being deployed to. As such, the cluster admin will need to configure the Prometheus Operator to watch for events in the correct namespace.
+
+### OpenShift Monitoring Prometheus Operator Support
+
+#### Requirements
+
+Once OLM offers bundle support for the `ServiceMonitor` and `PrometheusRule` resources, a sanctioned set of OLM managed operators should be able to report metrics to the OpenShift Monitoring Prometheus Operator. The OpenShift Monitoring Prometheus Operator has a few requirements that must be met for the in order for it to scrape an application's metrics.
+
+##### Namespace Requirements
+
+The application must be deployed in a namespace that:
+
+- Is prefixed with `openshift-`
+- Is labeled with `openshift.io/cluster-monitoring=true`
+
+##### ServiceMonitor Requirements
+
+A ServiceMonitor object must be created in the namespace mentioned above and point to the application's metrics endpoint.
+
+>Note: This requirement is already fulfilled with OLM bundle support for Prometheus resources.
+
+##### RBAC Requirements
+
+A role/rolebinding must be created that provides the Prometheus Operator ServiceAccount with the appropriate RBAC privileges to discover the `ServiceMonitor` and `PrometheusRule` resources in the newly created namespace.
+
+#### Fulfilling Namespace and RBAC Requirements
+
+To fulfill the requirements above, we propose adding an annotation to the `ClusterServiceVersion` whose key is `suggested-namespace` and whose value represent the namespace that the operator should be deployed to.
+
+The `suggested-namespace` annotation will only influence behavior when the operator is being installed via OpenShift Console UI. When the `suggested-namespace` annotation is present the UI will highlight the suggested namespace when installing the operator. OLM itself will not require that the operator is deployed in the namespace defined by the `suggested-namespace` annotation. This annotation will only be used by OLM to update the status of the CSV to highlight when an operator is not installed in a suggested namespace.
+
+If the user installs an operator through the UI in a namespace beings with `openshift-*` and the CSV has the `monitoringSupported` field set to `true`, the OpenShift Console will create the namespace with the required `openshift.io/cluster-monitoring=true` label. The UI will also be responsible for updating the OpenShift Monitoring Prometheus Operator ServiceAccount with the appropriate RBAC privileges for the given namespace as well.
```

Comment:
we probably are going to want a way to distinguish between wanting cluster monitoring and wanting user workload monitoring.

If the monitoring team updates cluster-monitoring to scrape any labeled namespace (not just openshift-* namespaces) as they have expressed a desire to do, then this will be very important so we're not just slapping the cluster-monitoring label on every namespace.

To that end can we name this field more explicitly?  `clusterMonitoringSupported` or something?

we may not need userMonitoringSupported field in the future since user workload monitoring is implied if your namespace isn't labeled for clustermonitoring, but i still think it's better to be explicit.
/cc @s-urbaniak @LiliC

Reply:
+1 on `clusterMonintoringSupported` as the field

Reply:
> we probably are going to want a way to distinguish between wanting cluster monitoring and wanting user workload monitoring.

Agreed.  `clusterMonitoringSupported` sgtm as weell, for clarity.